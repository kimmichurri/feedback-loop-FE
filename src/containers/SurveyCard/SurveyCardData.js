import React, { Component } from 'react';
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import UserAvg from './UserAverages'
import AvgResp from './AverageResponses'

export class SurveyCardData extends Component {
  constructor(props) {
    super(props);
    this.state = {
      averages: '',
      userAverages: ''
    }
  }

  componentDidMount = async () => {
    // avgsUrl = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.sData.id}/averages`
    // avgsResponse = await fetch(avgsUrl)
    // averages = await avgsResponse.json()
    //
    // userAvgsUrl = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.sData.id}/user_averages`
    // userAvgsResponse = await fetch(userAvgsUrl)
    // userAverages = await userAvgsResponse.json()

    this.setState({
      averages: AvgResp,
      userAverages: UserAvg
    })
  }


  displayGroup = (group) => {
    return(
      <section className="s-group">{group.name}
      </section>
    )
  }

  render() {
    return(
      <>
        <section className="s-status">{this.props.sData.status}</section>
        <section className="s-groups">{this.props.sData.groups.map(group => {
          return this.displayGroup(group)
        })}
        </section>
      </>
    )
  }
}

export default SurveyCardData
