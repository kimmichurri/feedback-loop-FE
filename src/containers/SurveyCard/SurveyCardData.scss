.s-status {
  align-self: flex-start;
  padding-bottom: 1%;
  color: $accent-charcoal;
}

.s-groups {
  margin-left: 10%;
  margin-right: -15px;
}

.s-group {
  padding-top: 2%;
  padding-bottom: 2%;
  border-top-left-radius: 12px;
  border-bottom-left-radius: 12px;
  background-color: white;
  color: black;
  box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
}

.s-group-name {
  text-align: left;
  padding-left: 2%;
}

.s-question, .user-rating, .q-avg-rating {
  text-align: end;
  padding-right: 2%;
}

.s-question-data, .s-question {
  margin-bottom: 1%;
}

.s-question {
  color: $accent-red;
  font-weight: bold;
}
