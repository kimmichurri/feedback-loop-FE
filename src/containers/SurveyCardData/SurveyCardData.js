import React, { Component } from 'react';
import { connect } from 'react-redux'
import { handleGet } from '../../thunks/handleGet'

export class SurveyCardData extends Component {
  constructor(props) {
    super(props);
    this.state = {
      averages: null,
      userAverages: null,
      survey: null
    }
  }

  componentDidMount = async () => {
    const avgsUrl = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.sData.id}/averages`
    const averages = await this.props.handleGet(avgsUrl)

    const userAvgsUrl = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/${this.props.sData.id}/user_averages`
    const userAverages = await this.props.handleGet(userAvgsUrl)

    this.setState({
      averages: averages,
      userAverages: userAverages,
      survey: this.props.sData
    })
  }

  displayGroup = (group) => {
    var group = group
    return(
      <section key={group.name} className="s-group">
        <section className="s-group-name group-box">
          Group {group.name}
        </section>
        {this.state.survey.questions.map(question => {
          return this.displayQuestionData(question, group)
        })}
      </section>
    )
  }

  displayQuestionData = (question, group) => {
    return(
      <section key={question.id} className="s-question-data">
        <article className="s-question group-box"> {question.questionTitle}</article>
        <article className='q-avg-rating group-box'>Survey Average: {this.state.averages.averages.length ? this.averageRating(question.id) : "Pending"}</article>
        <article className='u-ratings-container group-box'>
          {this.displayStudentData(question.id, group)}
        </article>
      </section>
    )
  }

  averageRating = (question_id) => {
    return(
      <>
        {this.state.averages.averages.map(average => {
            if (average.question_id === question_id) {
              return Number.parseFloat(average.average_rating).toFixed(2)
            }
          })
        }
      </>
    )
  }

  displayStudentData = (question_id, group) => {
    return(
      <section className='u-ratings'>
        {group.members.map(member => {
          var userAve = this.findUserAve(member.id, question_id)
          console.log("USERAVE: ", userAve)
          return <article className='user-rating'>
            {member.name}: {userAve ? userAve : "Pending"}
          </article>
        })}
      </section>
    )
  }

  findUserAve = (userId, question_id) => {
    var question_id = question_id
    var userId = userId
    let result
    this.state.userAverages.averages.map(average => {
      if (average.question_id == question_id && average.user_id == userId) {
        result = Number.parseFloat(average.average_rating).toFixed(2)
      }
    })
    return result
  }

  render() {
    return(
      <>
        {this.state.survey ?
          <>
            <section className="s-status">
              Survey Status:  {this.state.survey.status}
            </section>
            <section className="s-groups">
              {this.state.survey.groups.map(group => {
                return this.displayGroup(group)
              })}
            </section>
          </>
          :
          <p>NOT READY YET</p>
        }
      </>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handleGet: (url) => dispatch(handleGet(url))
})
export const mapStateToProps = (state) => ({
  error: state.error
})
export default connect(mapStateToProps, mapDispatchToProps)(SurveyCardData)
