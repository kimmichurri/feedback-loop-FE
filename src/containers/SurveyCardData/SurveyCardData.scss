.s-status {
  align-self: flex-start;
  padding-bottom: 1%;
  color: $accent-charcoal;
}

.s-groups {
  // margin-left: 10%;
  // margin-right: -15px;
  margin: auto;
  width: 90%;
}

.s-group {
  padding-top: 2%;
  padding-bottom: 2%;
  border-radius: 12px;
  // border-bottom-left-radius: 12px;
  background-color: white;
  color: black;
  box-shadow: 0 0 2px 1px rgba(0,0,0,.2);
}

.s-group-name {
  text-align: left;
  padding-left: 2%;
  padding-bottom: 2%;
}

.s-question, .user-rating, .q-avg-rating {
  text-align: end;
  padding-right: 2%;
}

.s-question-data, .s-question {
  margin-bottom: 1%;
}

.s-question {
  color: $accent-charcoal;
  font-weight: bold;
}

.q-avg-rating {
  font-style: italic;
}

.s-response-count {
  text-align: left;
}

.s-team-title {
  color: #2A2A2A;
  font-weight: bold;
}
