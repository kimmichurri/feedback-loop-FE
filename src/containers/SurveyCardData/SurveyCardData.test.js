import React from 'react'
import { shallow } from 'enzyme'
import { SurveyCardData, mapStateToProps, mapDispatchToProps } from './SurveyCardData'
jest.mock('../../thunks/handleGet')

describe('SurveyCardData', () => {
  let wrapper
  let mockHandleGet
  let mockError
  
  beforeEach(() => {
    mockHandleGet = jest.fn()
    mockError = "Something went wrong"
    wrapper = shallow(
      <SurveyCardData
        handleGet={mockHandleGet}
        error={mockError}
      />
    )
  })

  it ('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handleGet with a url on componentDidMount', () => {

  })

  describe('mapStateToProps', () => {

    it('should return an error as props', () => {
      const mockState = {
        error: "Something went wrong",
        fakeState: "fakeState"
      }
      const expected = {
        error: "Something went wrong"
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return handleGet as props', () => {
      const mockDispatch = jest.fn()
      const mockUrl = 'www.getSurveys.com'
      const handleGet = jest.fn()
      const actionToDispatch = handleGet(mockUrl)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.handleGet(mockUrl)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})