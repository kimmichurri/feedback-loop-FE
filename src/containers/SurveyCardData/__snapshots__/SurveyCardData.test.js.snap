// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyCardData should match the snapshot 1`] = `<Fragment />`;
